src/server/database/Database/Transaction.cpp

```diff
@@ -63,20 +67,51 @@ void Transaction::Cleanup()
     _cleanedUp = true;
 }
 
+std::string Transaction::ToString()
+{
+    std::string msg;
+    for (SQLElementData const& data : m_queries)
+    {
+        switch (data.type)
+        {
+            case SQL_ELEMENT_PREPARED:
+                msg += data.element.stmt->getQueryString();
+                break;
+            case SQL_ELEMENT_RAW:
+                msg += data.element.query;
+                break;
+        }
+        msg += ";";
+    }
+    return msg;
+}
+
 bool TransactionTask::Execute()
 {
     int errorCode = m_conn->ExecuteTransaction(m_trans);
     if (!errorCode)
+    {
+        m_trans->Cleanup();
```

Comment:
could you explain this change more in detail ?

Reply:
I have no idea about this, but I assumed it would be beneficial to free up the memory on even successful transactions. Is this not the case?

Reply:
I think this PR currently has few changes that are not needed. It's a good proof of concept, I'll polish and test it this weekend when I should have some free time

Reply:
https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/database/Database/Transaction.h#L39

Cleanup() is called automatically by the destructor